src/System.Text.Json/tests/Serialization/Array.ReadTests.cs

```diff
@@ -412,17 +411,32 @@ public static void ReadClassWithObjectImmutableTypes()
             obj.Verify();
         }
 
+        public class ClassWithPopulatedListAndNoSetter
+        {
+            public List<int> MyList { get; } = new List<int>() { 1 };
+        }
+
         [Fact]
         public static void ClassWithNoSetter()
         {
-            string json = @"{""MyList"":[1]}";
-            ClassWithListButNoSetter obj = JsonSerializer.Deserialize<ClassWithListButNoSetter>(json);
-            Assert.Equal(1, obj.MyList[0]);
+            // We don't attempt to deserialize into collections without a setter.
+            string json = @"{""MyList"":[1,2]}";
+            ClassWithPopulatedListAndNoSetter obj = JsonSerializer.Deserialize<ClassWithPopulatedListAndNoSetter>(json);
+            Assert.Equal(1, obj.MyList.Count);
         }
 
-        public class ClassWithListButNoSetter
+        public class ClassWithPopulatedListAndSetter
+        {
+            public List<int> MyList { get; set;  } = new List<int>() { 1 };
+        }
+
+        [Fact]
+        public static void ClassWithPopulatedList()
         {
-            public List<int> MyList { get; } = new List<int>();
+            // We don't attempt to deserialize into collections without a setter.
```

Comment:
Should this be something like "we replace the contents of this collection"?

Reply:
Thanks. I will change in future PR to avoid CI reset.